### DISCUSSION TITLE
Issues with custom auth

### DISCUSSION THREAD

**Discussion by van-vince** [refinedev/refine#3916] [Q&A] [answered]
I initiated my project with google auth, but now I want to use custom auth so i can get more control over the authentication, but the auth pages are not showing when I navigate to /login, /register etc. I dont know what is wrong cos its not showing any error in console too. I followed doc here.. https://refine.dev/docs/examples/authentication/mui/ 

update: apparently the logout button does not clear the localstorage so when i cleared it manually, the auth pages appeared. Now i need help in how to make logout button clear localstorage. This is the logout function in authProvider
logout: async () => {
localStorage.removeItem("email");
return {
success: true,
redirectTo: "/login",
};
},

**Comment by aliemir** [comment, marked as answer]
Hey @van-vince, by default, the logout button in `<Sider>` uses the `useLogout` hook and calls the `logout` method of the `authProvider`. Just like the example you've linked above, `logout` removes the local storage item and redirects to the path specified in `redirectTo`. 

Do you have a custom `<Sider>`? You might want to check if the `logout` method is called properly.

If you've switched from `v3` to v4` and used codemod, then switched your `authProvider`. I think you might have accidentally left `v3LegacyAuthProviderCompatible: true` in some of the authentication hooks.

When `v3LegacyAuthProviderCompatible` is set to `true`, then it will use the `legacyRouterProvider` prop rather than the `routerProvider` prop, which might lead to `logout` method not being called. 

Hope this resolves the issue 🙏

**Reply by van-vince** [reply]
Thanks
